Correct IPAM description (#117)
theforeman/foreman-documentation: guides/doc-Provisioning_Guide/topics/Networking.adoc

Subnet::
You must assign every host managed by {Project} to a subnet. Using subnets, {Project} can then manage IPv4 reservations. If there are no reservation integrations, you still must create and associate at least one subnet. When you manage a subnet in {Project}, you cannot create DHCP records for that subnet outside of {Project}. In {Project}, you can use IP Address Management (IPAM) to manage IP addresses with one of the following options:
+

> * *DHCP*: the DHCP {SmartProxy} manages the assignment of IP addresses by finding the next available IP address starting from the first address of the range skipping all addresses that are reserved. Before assigning an IP address, {SmartProxy} sends an ICMP and TCP ping to perform a brief check whether the IP address is in use. This check will not work for hosts which are turned off, so the DHCP option can be only used for subnets which are in full {Project} control without any hosts created externally. {SmartProxy} DHCP module retains offered IP addresses for short period of time to prevent collisions during concurrent access, therefore temporary "holes" in the IP range can exist.

Author: Foreman tries to "knock the doors" using PING command/protocol if the IP address is alive or not. The problem is, when a host with this IP assigned is literally powered off (turned off) or even (I missed this originally in my text) if it has a firewall configured in a way it would drop those packets, Foreman can make a false assumption the IP address is available.

Therefore users should be cautious and only use this for subnets which are under full control of Foreman.

> * *DHCP*: the DHCP {SmartProxy} manages the assignment of IP addresses by finding the next available IP address starting from the first address of the range skipping all addresses that are reserved. Before assigning an IP address, {SmartProxy} sends an ICMP and TCP ping to perform a brief check whether the IP address is in use. This check will not work for hosts which are turned off, so the DHCP option can be only used for subnets which are in full {Project} control without any hosts created externally. {SmartProxy} DHCP module retains offered IP addresses for short period of time to prevent collisions during concurrent access, therefore temporary "holes" in the IP range can exist.

Reviewer: And what are those subnets that are in full Foreman control? Does it mean that the subnet must be managed by Foreman only, on not by any other soft?

> * *DHCP*: the DHCP {SmartProxy} manages the assignment of IP addresses by finding the next available IP address starting from the first address of the range skipping all addresses that are reserved. Before assigning an IP address, {SmartProxy} sends an ICMP and TCP ping to perform a brief check whether the IP address is in use. This check will not work for hosts which are turned off, so the DHCP option can be only used for subnets which are in full {Project} control without any hosts created externally. {SmartProxy} DHCP module retains offered IP addresses for short period of time to prevent collisions during concurrent access, therefore temporary "holes" in the IP range can exist.

Author: Yes not by other software or person (manually modifying configuration or database files).

Subnet::
You must assign every host managed by {Project} to a subnet. Using subnets, {Project} can then manage IPv4 reservations. If there are no reservation integrations, you still must create and associate at least one subnet. When you manage a subnet in {Project}, you cannot create DHCP records for that subnet outside of {Project}. In {Project}, you can use IP Address Management (IPAM) to manage IP addresses with one of the following options:
+

> * *DHCP*: DHCP {SmartProxy} manages the assignment of IP addresses by finding the next available IP address starting from the first address of the range skipping all addresses that are reserved. Before assigning an IP address, {SmartProxy} sends an ICMP and TCP pings to check whether the IP address is in use. Note that if a host is powered off, or has a firewall configured to disable connections, {Project} makes a false assumption that the IP address is available. Therefore, use the *DHCP* option with subnets that are in full {Project} control and that do not have hosts created externally. This check does not work for hosts that are turned off, so the DHCP option can only be used for subnets that {Project} controls and that do not have any hosts created externally.

Reviewer 2: ```suggestion
* *DHCP*: DHCP {SmartProxy} manages the assignment of IP addresses by finding the next available IP address, starting from the first address of the range and skipping all addresses that are reserved. Before assigning an IP address, {SmartProxy} sends an ICMP and TCP pings to check whether the IP address is in use. Note that if a host is powered off, or has a firewall configured to disable connections, {Project} makes a false assumption that the IP address is available. Therefore, the *DHCP* option can only be used with subnets that {Project} controls and that do not have any hosts created externally. 
```
The last sentence was redundant, just repeating the sentences before it.

+

> {SmartProxy} DHCP module retains offered IP addresses for a short period of time to prevent collisions during concurrent access, therefore some IP addresses in the IP range might remain temporary unused.

Reviewer 2: ```suggestion
The {SmartProxy} DHCP module retains the offered IP addresses for a short period of time to prevent collisions during concurrent access, so some IP addresses in the IP range might remain temporarily unused.
```

> * *Internal DB*: {Project} finds next available IP address from the Subnet range excluding all IP addresses from the {Project} database in sequence. Primary source of data is the database, not DHCP reservations. This IPAM is not safe when multiple hosts are being created in parallel, in that case use *DHCP* or *Random DB* IPAM instead.

Reviewer 2: ```suggestion
* *Internal DB*: {Project} finds the next available IP address from the Subnet range by excluding all IP addresses from the {Project} database in sequence. The primary source of data is the database, not DHCP reservations. This IPAM is not safe when multiple hosts are being created in parallel; in that case, use *DHCP* or *Random DB* IPAM instead.
```

> * *Random DB*: {Project} finds next available IP address from the Subnet range excluding all IP addresses from the {Project} database randomly. Primary source of data is the database, not DHCP reservations. This IPAM is safe to use with concurrent host creation as IP addresses are returned in random order minimizing chance of a conflict.

Reviewer 2: ```suggestion
* *Random DB*: {Project} finds the next available IP address from the Subnet range by excluding all IP addresses from the {Project} database randomly. The primary source of data is the database, not DHCP reservations. This IPAM is safe to use with concurrent host creation as IP addresses are returned in random order, minimizing the chance of a conflict.
```

* *EUI-64*: Extended Unique Identifier (EUI) 64bit IPv6 address generation, as per RFC2373, is obtained through the 48-bit MAC address.

> * *External IPAM*: Delegate IPAM to an external system through {SmartProxy} feature. {Project} currently does not ship with any external IPAM implementations, several plugins are in development.

Reviewer 2: ```suggestion
* *External IPAM*: Delegates IPAM to an external system through the {SmartProxy} feature. {Project} currently does not ship with any external IPAM implementations, but several plug-ins are in development.
```
Should we exclude this one from the downstream build?
